lib/ansible/galaxy/collection.py

```diff
@@ -584,6 +591,82 @@ def build_collection(collection_path, output_path, force):
     return collection_output
 
 
+def _collection_matches_requirement(collection, requirement, parent=None):
```

Comment:
Is it worth having this in a separate function considering it's only used in 1 place?